### PR TITLE
planning docs

### PR DESCRIPTION
**Pull request by daniel-noland**
image

Unfortunately this commit required changes to dpdk-sys to build.  As a result, we lost nightly-sterile builds in a commit where I normally would not have messed with that.

### CODE REVIEW

**Comment by qmonnet** [comment]
It builds but the preview is broken for me, the link to most pages are invalid

File: design-docs/src/mdbook/src/links.md
Comment (qmonnet) [inline design-docs/src/mdbook/src/links.md:1]: This file currently seems broken, the rendered page looks empty

File: design-docs/src/mdbook/src/dataplane/design-session.md [hunk 1]
```diff
@@ -0,0 +1,393 @@
...
+This is a component that exchanges [netlink] messages with the [kernel] in response to changes in the [dataplane model].
+Its responsibilities include
+
+1. construction of virtual network interfaces needed by [zebra]
+2. translation of ephemeral linux kernel parameters into ephemeral [dpdk] parameters (e.g. netlink interface index to dpdk interface id).
+3. retrieval of information not available to [zebra]/[frr] such as neighbor tables / [ARP] / [IPv6 ND] resolution or [bridge] fdb.
+
+### Control Plane Interface
+
+This component is responsible for adjudicating communication between the [control plane] and the [dataplane].
```
Comment (qmonnet) [inline design-docs/src/mdbook/src/dataplane/design-session.md:315]: The `[dataplane]` is referenced multiple times, but it doesn't seem to have a corresponding entry; the links don't take me anywhere

File: design-docs/src/mdbook/src/links.md
Comment (daniel-noland) [reply design-docs/src/mdbook/src/links.md:1]: will fix

File: design-docs/src/mdbook/src/dataplane/design-session.md (hunk 1 above)
Comment (daniel-noland) [reply design-docs/src/mdbook/src/dataplane/design-session.md:315]: yup, will fix

File: design-docs/src/mdbook/src/links.md
Comment (daniel-noland) [reply design-docs/src/mdbook/src/links.md:1]: should be resolved

**Review by qmonnet — CHANGES_REQUESTED** [review]
The changes look good, but please split-up this PR into logical commit (or PRs?). The change in CI, in source code, or in the build framework seem unrelated to the doc additions.

**Comment by daniel-noland** [comment]
> The changes look good, but please split-up this PR into logical commit (or PRs?). The change in CI, in source code, or in the build framework seem unrelated to the doc additions.

This should be resolved now.  I appreciate the attention to detail.

File: design-docs/src/mdbook/src/dataplane/tasks/zebra-plugin.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+4. multicast routes (this is unlikely to be important in the near term).
+
+We only expect to support basic IPv4 and IPv6 LPM routes in the near term, but feature evolution should be accounted for in the design.
+
+## Likely dispatch
+
+* [@Fredi-raspall]
+* coordinate with: [@daniel-noland]
+
+{{#include ../../links.md}}
```
Comment (qmonnet) [inline design-docs/src/mdbook/src/dataplane/tasks/zebra-plugin.md:24]: I know that with Sphinx, there's [an option](https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-rst_epilog) to include this sort of stuff automatically in the footer of all pages. I don't know if mdBook (on a quick look, I couldn't find an equivalent option). Anyway, doesn't matter much.

**Review by qmonnet — APPROVED** [review]
Looks so much cleaner now, thank you! :+1:

**Comment by qmonnet** [comment]
Fine by me, you can merge it or wait for more reviews if your prefer

File: design-docs/src/mdbook/src/dataplane/tasks/zebra-plugin.md (hunk 2 above)
Comment (daniel-noland) [reply design-docs/src/mdbook/src/dataplane/tasks/zebra-plugin.md:24]: Yeah, I could make it work.  I just didn't try that hard for such a small thing

**Pull request merged**